### ISSUE TITLE
Not sure about this, I don't know the internals enough, mime_pack/3 is not deterministic but documentation claims it is det???

### ISSUE THREAD

**Issue by xpxaxsxi** [SWI-Prolog/packages-http#91]
If you copy this swish-file to a local Swi-prolog and call tmp1:tst/2 
[http://swish.swi-prolog.org/p/rjJRlaZl.pl](http://swish.swi-prolog.org/p/rjJRlaZl.pl) 
I get a choicepoint.

Predicate pack(html(HTML), Out) is missing a cut?????? 

I am running threaded, 32 bits, version 7.5.2 on a Windows 10

**Comment by xpxaxsxi** [comment]
Fatal error with  SWI-Prolog (threaded, 64 bits, version 8.1.0)  Windows 10 Home

In 2017 I wrote a add-on (using some hook) to http_post that instead of loading a huge file to memory it sends a file from a disk. Back then I noticed that  mime_pack/3 does leave a choice point, I fixed that with once(mime_pack(_,_,_)) 

I am not using my add-on now. Because it was sort of hackish code, I was not sure if I found a bug then. But when I today returned to this Issue and decided to try it out, I am now getting a fatal error  (but I don't know if error is related to mime_pack/3 ) 

I will download a latest build tomorrow possibly and try again. 

1) `-? consult(user).` with 
```
:- module(tmp1,[]).
:- use_module(library(http/mimepack)).

tst:-
    current_output(Out),
    mime_pack:mime_pack([],Out,[mime_pack(html([text]))]).


tst2:-
    current_output(Out),
    A=test,
    mime_pack:mime_pack(
                  [html([text]),
                   hello_world,A,one,two],Out,_).

```

2) copy&paste it to Swipl, then hit return and ctrl-d
3) `gtrace.`
4) `[trace] -? tst2.`
`Correct to: "tmp1:tst2"? y `

5) In the tracer window , hit F (finish)
6) In Call stack window, click pack/2
7) [FATAL ERROR: at Thu Mar 28 20:50:11 2019
      Received signal 22 (abrt) while in 31-th garbage collection]
